src/chains/eos_2/eosCreateAccount.ts

```diff
@@ -93,6 +102,48 @@ export class EosCreateAccount implements CreateAccount {
     return null
   }
 
+  get isMultisig() {
+    if (isNullOrEmpty(this.options?.multisigOptions)) return false
+    return true
+  }
+
+  assertValidOptionsAndSetAuth() {
+    const { publicKeys, multisigOptions } = this.options
+    // If authorizations passed, throw if any other auth options is passed (publicKeys or multisigOptions) and return
+    if (!isNullOrEmpty(publicKeys) && !isNullOrEmpty(multisigOptions))
+      throwNewError('Both publicKeys and multisigOptions can not be provided')
+
+    if (!isNullOrEmpty(publicKeys)) {
+      this._authorizations.owner = publicKeys?.owner ? publicKeyToAuth(publicKeys.owner) : null
+      this._authorizations.active = publicKeys?.active ? publicKeyToAuth(publicKeys.active) : null
+      return
+    }
+
+    if (!isNullOrEmpty(multisigOptions)) {
+      const { owner, active } = multisigOptions
+      if (isNullOrEmpty(owner) && isNullOrEmpty(active)) {
+        throwNewError('If multisigOptions is provided, both active and owner has to be provided.')
+      }
+      this._authorizations.owner = isValidEosPublicKey(multisigOptions?.owner as EosPublicKey)
+        ? publicKeyToAuth(multisigOptions.owner as EosPublicKey)
+        : msigOptionToAuth(multisigOptions.owner as MsigAuthOptions)
+      this._authorizations.active = isValidEosPublicKey(multisigOptions?.active as EosPublicKey)
+        ? publicKeyToAuth(multisigOptions.active as EosPublicKey)
+        : msigOptionToAuth(multisigOptions.active as MsigAuthOptions)
+    }
+  }
+
+  /** Only available if NOT multisig */
+  get publicKeys(): EosPublicKeys {
+    if (this.isMultisig) {
+      throwNewError('get publicKeys can NOT be called if multisigOptions is provided.')
```

Comment:
Why is this the case? Couldn't this just return an array of keys to owner and active? 
Something like:
```
     owner: this._authorizations?.owner?.keys,
      active: this._authorizations?.active?.keys,
```
I might be missing the purpose of this function but it seems we also really want to return all permissions not just owner and active

Reply:
> It seems the creating and signing of multisig are at odds with each other. It looks like we only support creating an account with existing accounts and signing only supports signing with just keys. Am I missing something?

creating account supports both plain publicKeys and accounts(with permission name). Signing of course requires private key regardless if multisig account permissions consist of only publicKey or account/permission tuple.

Reply:
> Why is this the case? Couldn't this just return an array of keys to owner and active? Something like:
> 
> ```
>      owner: this._authorizations?.owner?.keys,
>       active: this._authorizations?.active?.keys,
> ```
> 
> I might be missing the purpose of this function but it seems we also really want to return all permissions not just owner and active

Now instead of passing publicKeyOwner and publicKeyActive into native eos account creation, we convert publicKeys or multisigOptions info provided from transaction options into this._authorization and pass ownerAuth and activeAuth into create native account template. get publicKeys() getter is created for using with virtual or createescrow account creations which only supports plain publicKey for active and owner permissions. Its also used with generateKeysIfNeeded, which is only needs to be called if account is not multisig. If multisigOptions passed account creation has to be native and we require all permissions specified, thus there is no use for get publicKeys.

Reply:
> creating account supports both plain publicKeys and accounts(with permission name). Signing of course requires private key regardless if multisig account permissions consist of only publicKey or account/permission tuple.

I think I misspoke so let me try again. 

I do not believe the example that you give for creating msig accounts will not work with the way msig signing has been implemented see `hasSignatureForAuthorization`. Only the `keys` portion is checked because checking for accounts is a whole other problem. 

Basically we should just show creating an account with keys and not accounts since chainjs cannot currently do anything with a msig account created with accounts.

Reply:
We could do that but the reason I didn't go down that rabbit hole is because you could have n number of accounts. For example:

accountA:
   active: accountB@active
   
accountB:
   active: accountC@active

accountC:
   active: accountD@active
   
   
 So as far as I can see a deeper discussion about what we want to support is required because in theory the nesting can go on for a while and that could cause performance issues as well as just tons of calls to apis to resolve keys. Plus we haven't even talked about what if the permission is itself a msig? 
 
 In my head the first pass should just show keys and we can design a good solution for supporting accounts.

Reply:
@traylewin thoughts?

Reply:
I think supporting that and using it on oreVault provides great value, giving allowing jointAccount owners to take ownership of their chainAccount and still be one of the owner of the joint account. Since the accountName is the real address on EOS paradigm, it would also be consistent with other chains in my opinion. 

As far as the nested permission issue goes, its just possible to overwhelm apis in other ways like just appending a lot of permissions to a single action, or trying to sign a transaction with a lot of actions in it. So to me, not supporting it doesn't add any benefit. 

Also to note, this is not just a multisig issue. Currently chain-js as a generic library does not work with accounts that have accountName/permissionName attached to its permissions. Hardcoded to expect keys array.

Reply:
@deckb I updated eosTransaction class to support accounts[] field IF the provided accountName/permission tuple for authorization does not contain another accounts[] for its authorization (hence nested accounts[] auth is not supported for now, and throws error). I think this provides the support without getting things too complicated and prone to execution errors. And it will allow us to use oreVault the way we initially decided with @traylewin